cmd/mavenBuild.go

```diff
@@ -0,0 +1,60 @@
+package cmd
+
+import (
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/maven"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+)
+
+func mavenBuild(config mavenBuildOptions, telemetryData *telemetry.CustomData) {
+	c := command.Command{}
+
+	c.Stdout(log.Entry().Writer())
+	c.Stderr(log.Entry().Writer())
+
+	err := runMavenBuild(&config, telemetryData, &c)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runMavenBuild(config *mavenBuildOptions, telemetryData *telemetry.CustomData, command execRunner) error {
+	var flags = []string{"-update-snapshots", "--batch-mode"}
+
+	exists, _ := piperutils.FileExists("integration-tests/pom.xml")
```

Comment:
Just a minor thing: You cannot test this code. I am not sure if it is the best solution, but you can see a pattern [here](https://github.com/SAP/jenkins-library/blob/it/nexus-upload-cmd/cmd/nexusUpload.go#L22-L63). Another option would be to define a small struct that consists of pointers to instances implementing certain interfaces (piperutils.Fileutils and execRunner). In the test, it is then easy pass an instance to runMavenBuild with mock-implementations.